### ISSUE TITLE
Wrong physical sizes in `meta.yml` of container

### ISSUE THREAD

**Issue by pastewka** [ContactEngineering/TopoBank#685]
I've just implemented a check (when reading a container) that the physical sizes in `meta.yml` are consistent with the data obtained from the reader. When reading https://contact.engineering/go/8sc7t/ I get the following error:

```
>>> from SurfaceTopography import read_container
>>> c, = read_container('surface.zip')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/pastewka/Quelltexte/SurfaceTopography/SurfaceTopography/Container/IO.py", line 123, in read_container
    raise ValueError(f'Physical sizes from data file (={physical_sizes_from_file} and from '
ValueError: Physical sizes from data file (=(0.29878356133333334,) and from meta.yml (=(0.298684,)) differ for topography uncd_stylus_300um_5mar_3_k0HpKJt.txt
```

Is it possible, that there is a pipeline function that changes the physical sizes?

**Comment by mcrot** [comment]
With `read_container` from SurfaceTopography version 0.94.0 and 0.94.1.dev5+ge2835b32 (current master) I get:
```python
[...]

~/work/imtek/topobank/venv/lib/python3.8/site-packages/SurfaceTopography/IO/NC.py in topography(self, channel_index, physical_sizes, height_scale_factor, info, periodic, subdomain_locations, nb_subdomain_grid_pts)                                                               
    268             else:                                                                                                                                                               
    269                 if physical_sizes is not None:                            
--> 270                     raise ValueError('You cannot specify physical sizes for a nonuniform topography.')                                                                          
    271                                                                                     
    272                 if self._periodic:                                                  
                                              
ValueError: You cannot specify physical sizes for a nonuniform topography.
```

Do you still have the output as in the issue description?

**Comment by pastewka** [comment]
Don't use master, use the `21_scale_dependent_...` branch

**Comment by mcrot** [comment]
Thanks, I've had a look. The value 0.298684 in the meta.yaml corresponds to the last data point in the text file.
The x values here range from 0.000000 to 0.298684.
For the value in the meta file of the container, I've used the `size_x` value saved in the database at the time
the surface was exported. It was once set by the size found from the file contents.

When I upload the same file **now** than 0.29878356133333334 is chosen for the size.
This is also what the channel reports.

Question: Is it possible, that in an earlier version of SurfaceTopography 0.298684 was reported as size?
Has the way the size for `UniformLineScanAndTopography` has changed in `SurfaceTopography`?

**Comment by pastewka** [comment]
It shouldn't have changed but I won't rule that out. If it works now, let's shelve this issue.

**Comment by mcrot** [comment]
This may also be related to #690. What seems to be needed here is to refresh the sizes of the topographies in the database.
This should be done after installing the fix for #690 and recreation of squeezed data files. 
I'll implement a management command for resetting the sizes and move this issue to the next version.
If the issue is solved then, it can be closed.

**Comment by mcrot** [comment]
Now, after 0.14.0 being installed, all squeezed files recreated and all sizes fixed, the surface can be downloaded again and this issue doesn't occur any more. Closing.
